Source/Core/DolphinQt/Achievements/AchievementBox.cpp

```diff
@@ -35,13 +37,18 @@ AchievementBox::AchievementBox(QWidget* parent, rc_client_achievement_t* achieve
   QSizePolicy sp_retain = m_progress_bar->sizePolicy();
   sp_retain.setRetainSizeWhenHidden(true);
   m_progress_bar->setSizePolicy(sp_retain);
+  m_progress_label = new QLabel();
+  m_progress_label->setAlignment(Qt::AlignCenter);
 
   QVBoxLayout* a_col_right = new QVBoxLayout();
   a_col_right->addWidget(title);
   a_col_right->addWidget(description);
   a_col_right->addWidget(points);
   a_col_right->addWidget(m_status);
   a_col_right->addWidget(m_progress_bar);
+  QVBoxLayout* a_prog_layout = new QVBoxLayout(m_progress_bar);
+  a_prog_layout->setContentsMargins(0, 0, 0, 0);
+  a_prog_layout->addWidget(m_progress_label);
```

Comment:
Err, you're not doing anything with the layout here...?

Reply:
This is what overlays the label on top of the progress bar.
<img width="407" alt="image" src="https://github.com/dolphin-emu/dolphin/assets/122337285/6e883cba-6904-436c-b4c7-2995e4cf2b7a">

Reply:
But don't you have to add the layout to some container or something? Or does the constructor do that implicitly? Why is Qt so weird with memory management??

Reply:
My understanding is that by passing m_progress_bar into its constructor as its parent then it adds the layout to m_progress_bar? I think.

Reply:
Okay yeah looks like `new QVBoxLayout(x)` is equivalent to `x.setLayout(new QVBoxLayout())`. I guess this is fine then, even if it looks like a memory leak...